File: Leetcode/04_most_common_word.py
from typing import List
from collections import *
from re import sub


def mostCommonWord(paragraph: str, banned: List[str]) -> str:
    words = [
        word
        for word in sub(r"[^\w]", " ", paragraph).lower().split()
        if word not in banned
    ]

    # counts = defaultdict(int)
    # for word in words:
    #     counts[word] += 1

    # return max(counts, key=counts.get)

    return Counter(words).most_common()[0][0]


print(
    mostCommonWord("Bob hit a ball, the hit BALL flew far after it was hit.", ["hit"])
)

File: Leetcode/04_most_common_word.js
/**
 * @param {string} paragraph
 * @param {string[]} banned
 * @return {string}
 */
var mostCommonWord = function (paragraph, banned) {
  let dict = {};
  let filterParagraph = paragraph
    .toLowerCase()
    .match(/[a-zA-Z]+/g)
    .filter((v) => !banned.includes(v));

  filterParagraph.forEach((v) => {
    if (!dict[v]) {
      dict[v] = 1;
    } else {
      dict[v] += 1;
    }
  });

  return Object.keys(dict).reduce((p, c) => (dict[p] > dict[c] ? p : c));
};

console.log(
  mostCommonWord('Bob hit a ball, the hit BALL flew far after it was hit.', [
    'hit',
  ]),
);
